src/weapon/w_011.c

```diff
@@ -155,8 +158,218 @@ int GetWeaponId(void) { return 11; }
+            xOffset = 8;
+            if (self->velocityX < 0) {
+                xOffset = -xOffset;
+            }
+            yOffset = 22;
+            xVar = self->posX.i.hi + xOffset;
+            yVar = self->posY.i.hi + yOffset;
+            g_api.CheckCollision(xVar, yVar, &collider, 0);
+            if ((collider.effects & EFFECT_SOLID) &&
+                !(collider.effects & EFFECT_UNK_8000)) {
+                goto make_other_entity;
+            }
+            xOffset = 8;
+            if (self->velocityX < 0) {
+                xOffset = -xOffset;
+            }
+            yOffset = 8;
+            xVar = self->posX.i.hi + xOffset;
+            yVar = self->posY.i.hi + yOffset;
+            g_api.CheckCollision(xVar, yVar, &collider, 0);
+            if ((collider.effects & EFFECT_SOLID) &&
+                !(collider.effects & EFFECT_UNK_8000)) {
+                goto make_other_entity;
+            }
+        }
+        if ((((collider.effects & (EFFECT_UNK_8000 | EFFECT_UNK_4000)) ==
+              (EFFECT_UNK_8000 | EFFECT_UNK_4000)) &&
+             (self->facingLeft)) ||
+            (((collider.effects & (EFFECT_UNK_8000 | EFFECT_UNK_4000)) ==
+              EFFECT_UNK_8000) &&
+             (self->facingLeft == 0))) {
+            if (!(collider.effects &
+                  (EFFECT_SOLID_FROM_ABOVE | EFFECT_SOLID_FROM_BELOW))) {
+                SetSpeedX(FIX(1.25));
+            }
+            // @bug: These next two if-statements are never true because the bit
+            // flags don't match up. We're testing against 0x40 and 0x80 flags,
+            // and seeing if they match 0x100 or 0x200.
+            if ((collider.effects &
+                 (EFFECT_SOLID_FROM_ABOVE | EFFECT_SOLID_FROM_BELOW)) ==
+                EFFECT_UNK_0100) {
+                SetSpeedX(FIX(2));
+            }
+            if ((collider.effects &
+                 (EFFECT_SOLID_FROM_ABOVE | EFFECT_SOLID_FROM_BELOW)) ==
+                EFFECT_UNK_0200) {
+                SetSpeedX(FIX(2.25));
+            }
+        }
+        xOffset = 8;
+        if (self->velocityX < 0) {
+            xOffset = -xOffset;
+        }
+        yOffset = -8;
+        xVar = self->posX.i.hi + xOffset;
+        yVar = self->posY.i.hi + yOffset;
+        g_api.CheckCollision(xVar, yVar, &collider, 0);
+        if (!(collider.effects & EFFECT_SOLID) && (!self->hitFlags)) {
+            if (self->ext.timer.t % 4 == 0) {
+                // Blueprint 90 is weapon entity 9, func_ptr_80170024, so
+                // ourself.
+                unusedEnt = g_api.CreateEntFactoryFromEntity(
+                    self, ((g_HandId + 1) << 0xE) + FACTORY(0x100, 90), 0);
+            }
+            self->posX.val += self->velocityX;
+        } else {
+            goto make_other_entity;
+        }
+        break;
+    case 3:
+        if (self->animFrameDuration >= 0) {
+            break;
+        }
+        DestroyEntity(self);
+        return;
```

Comment:
```suggestion
        if (self->animFrameDuration < 0) {
            DestroyEntity(self);
            return;
        }
            break;
```
flipping this for consistency. I did not test if this matches

Reply:
Ah yes good point of course, will swap